### PR TITLE
Fix sub string operation made string none-utf8

### PR DESCRIPTION
**Pull request by hocktea214**
Signed-off-by: zheng <huxiangyuan@harmonycloud.cn>

### CODE REVIEW

File: collector/analyzer/tools/utf8.go [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+}
+
+func runeIndex(p []byte) int {
+	np := len(p)
+	var n int
+	for i := 0; i < np; {
+		c := p[i]
+		if c < utf8.RuneSelf {
+			// ASCII fast path
+			if n > 0 && n < 3 {
```
Comment (dxsup) [inline collector/analyzer/tools/utf8.go:72]: Why is it 3 here? Could you add some comments to state the reason?

Comment (hocktea214) [reply collector/analyzer/tools/utf8.go:72]: As the size of chinese character is 3, we only support it for now.

Comment (hocktea214) [reply collector/analyzer/tools/utf8.go:72]: > Why is it 3 here? Could you add some comments to state the reason?
As the size of chinese character is 3, we only support it for now.

Comment (hocktea214) [reply collector/analyzer/tools/utf8.go:72]: > Why is it 3 here? Could you add some comments to state the reason?

As the size of chinese character is 3, we only support it for now.

File: collector/analyzer/tools/utf8.go [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+		} else if c := s[i+3]; c < locb || hicb < c {
+			size = 1
+		}
+		n = size
+		i += size
+	}
+	return ns
+}
+
+func FomratStringToUtf8(s string) string {
```
Comment (dxsup) [inline collector/analyzer/tools/utf8.go:158]: I think `Fomrat` is a typo which is supposed to be `Format`.

**Comment by dxsup** [comment]
Since the part of file `utf8.go` is copied from the standard library `unicode/utf8`, I think we should include the license text at the head of the file.

File: collector/analyzer/tools/utf8.go (hunk 2 above)
Comment (hocktea214) [reply collector/analyzer/tools/utf8.go]: Wrong spell word, check it.

Comment (hocktea214) [reply collector/analyzer/tools/utf8.go]: It‘s a spelling mistake, fix it.

**Review by dxsup — APPROVED** [review]

**Pull request merged**